src/libostree/ostree-sysroot.c

```diff
@@ -1645,22 +1678,41 @@ ostree_sysroot_simple_write_deployment (OstreeSysroot      *sysroot,
        * deployments, fall back on merge deployment */
       const gboolean passed_crossover = booted_deployment ? !before_booted : !before_merge;
 
+      gboolean is_previous_version = FALSE;
+      if (is_booted || is_merge)
+        {
+          g_autofree char *deployment_version =
+            ostree_deployment_get_version (sysroot, deployment);
+
+          if (!last_version)
+            last_version = g_steal_pointer (&deployment_version);
+
+          is_previous_version = (osname_matches && passed_crossover && last_version &&
+                                 (g_strcmp0 (deployment_version, last_version) != 0));
```

Comment:
Hmm, we should be checking this on every iteration, not just when it's the booted or merge deployment, no? Otherwise, it'll never trigger.

Reply:
Yes, what I want is to stop checking versions once I've already retained a previous version.